### PR TITLE
Enh/Fix: make modules still compatible with us

### PR DESCRIPTION
**Pull request by gst**
This one should be better than https://github.com/Alignak-monitoring/alignak/pull/26

because this one is dynamic (and so does not preload/import every possible module).

### CODE REVIEW

**Comment by gst** [comment]
@Seb-Solon  ..

which one do you prefer ?

**Comment by Seb-Solon** [comment]
This one is better IMO

File: alignak/shinken_import_hook.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+import importlib
+import sys
+
+
+class Finder(object):
+
+    def find_module(self, fullname, path=None):
+        if fullname == 'shinken' or fullname.startswith('shinken.'):
+            return self
```
Comment (ddurieux) [inline alignak/shinken_import_hook.py]: module with shinken ?

**Comment by ddurieux** [comment]
Not really why you do this?
You have problem on load alignak modules?

**Comment by aviau** [comment]
We don't want to make our modules incompatible with shinken.

**Comment by Seb-Solon** [comment]
shinken module usually does from shinken import blabla. That's the issue. This comes from the fact you change the folder and everything to alignak when renaming instead of only editing header. We have to adapt the code now to be able to load shinken modules.

**Comment by ddurieux** [comment]
I understand that, but:

* For the moment shinken and alignak have same structure, in future no
* Alignak will be in python 2.7 and 3.4 (currently in progress), so all shinken module will be compatible
* Modules for alignak will be installed only with 'pip install', not the case with shinken 
* Modules fro broker may be all in external (internal is deleted, work in progress), perhaps can do same with other modules
* We want too some module dropped because have too many module is not a good idea. You can see all retntion modules of shinken and more than half not work

**Comment by Seb-Solon** [comment]
I don't get it. This PR was a technical fix because otherwiase shinken module can't work with alignak (and vice versa) 
The fact all module have a import shinken / import alignak force us to hack this.
Otherwise, we have to right now fork all module we need and replace those import. 
This PR is to keep everything working for now (because @aviau for example may need it). Of course we should remove this later. 

Point 2 3 and 4 are not related to this PR IMO.

**Comment by ddurieux** [comment]
You don't think it's like open the Pandora's box ?

**Comment by aviau** [comment]
> You don't think it's like open the Pandora's box ?

Is there any downsides to making modules compatible?

**Comment by ddurieux** [comment]
it will not be complex (mean problems to maintain, specific errors, specific features) with time and divergence of code between alignak and shinken ?

**Comment by aviau** [comment]
Do we plan on heavily changing the modules API?

If yes, are we doing it right now?

Until this happens, why not use the same modules?

**Comment by gst** [comment]
> I understand that, but:
> ...

that's all right I think.

This PR is only for being able to run, without a change at them, shinken modules *up to / until* we'll change this API (if we need to, not impossible.).

**Comment by gst** [comment]
So ok for merge ? who wants ?

**Comment by ddurieux** [comment]
I'm ok to merge but before, can you add an enhance issue with these points: 

* ref to this PR to clean later
* add comment to delete shinken.io ref in code (because modules will be installer via pip install).

**Pull request merged**